golden/runner.py

```
@@ -213,6 +215,167 @@ def _format_stale_paths(stale: list[Path], work_dir: Path, max_show: int = 5) ->
+    except ImportError:
+        return
+    if hasattr(Worker, "chip_contexts"):
+        chip_contexts_installed = True
+    else:
+        chip_contexts_installed = False
+
+    if not chip_contexts_installed:
+        def _chip_contexts(self: Any) -> list[Any]:  # noqa: ANN001 - runtime compatibility shim
+            return []
+
+        setattr(Worker, "chip_contexts", property(_chip_contexts))
```

Ruff (0.15.18)

[warning] Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)